src/extension.ts

```diff
@@ -5,13 +5,14 @@ import * as server from './server';
 import { RSPClient } from 'rsp-client';
 import { ExtensionAPI, CommandHandler } from './extensionApi';
 
-const client = new RSPClient('localhost', 27511);
+let client = new RSPClient('localhost', 27511);
```

Comment:
No need to create server here, just check if server started before shutting it down in deactivate method.